### PR TITLE
Rework filtering spawn egg and tile entity nbt config

### PR DESCRIPTION
**Pull request by Machine-Maker**

### CODE REVIEW

File: patches/server/0807-config-for-disabling-entity-tag-tags.patch [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+diff --git a/src/main/java/net/minecraft/world/entity/EntityType.java b/src/main/java/net/minecraft/world/entity/EntityType.java
+index 1c446dba5de89698397041ee38a2e1a00bec8a56..6a8d4fd1adc5d931f91696ac7680109f418ec268 100644
+--- a/src/main/java/net/minecraft/world/entity/EntityType.java
++++ b/src/main/java/net/minecraft/world/entity/EntityType.java
+@@ -424,6 +424,18 @@ public class EntityType<T extends Entity> implements EntityTypeTest<Entity, T> {
+                 if (world.isClientSide || !entity.onlyOpCanSetNbt() || player != null && minecraftserver.getPlayerList().isOp(player.getGameProfile())) {
+                     CompoundTag nbttagcompound1 = entity.saveWithoutId(new CompoundTag());
+                     UUID uuid = entity.getUUID();
++                    // Paper start - filter out protected tags
++                    if (player == null || !minecraftserver.getPlayerList().isOp(player.getGameProfile())) {
```
Comment (Proximyst) [inline patches/server/0807-config-for-disabling-entity-tag-tags.patch]: Should use permissions

File: patches/server/0807-config-for-disabling-entity-tag-tags.patch [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
++                        net.minecraft.commands.arguments.NbtPathArgument nbtPathArgument = net.minecraft.commands.arguments.NbtPathArgument.nbtPath();
++                        for (String tag : world.paperConfig.disabledEntityTagTags) {
++                            try {
++                                nbtPathArgument.parse(new com.mojang.brigadier.StringReader(tag)).remove(itemNbt.getCompound("EntityTag"));
```
Comment (Proximyst) [inline patches/server/0807-config-for-disabling-entity-tag-tags.patch]: This seems real wasteful...

**Review by Proximyst — CHANGES_REQUESTED** [review]

File: patches/server/0807-config-for-disabling-entity-tag-tags.patch (hunk 1 above)
Comment (Machine-Maker) [reply patches/server/0807-config-for-disabling-entity-tag-tags.patch]: Now using `minecraft.nbt.place`

File: patches/server/0807-config-for-disabling-entity-tag-tags.patch (hunk 2 above)
Comment (Machine-Maker) [reply patches/server/0807-config-for-disabling-entity-tag-tags.patch]: Yep, it certainly is. Moved the `NbtPath` creating to PaperWorldConfig so it happens once. Has the benefit of notifying users of an invalid tag at startup as well.

**Comment by Machine-Maker** [comment]
Rebased for 1.18

**Comment by Machine-Maker** [comment]
Rebased for 1.19.3

**Review by jpenilla — APPROVED** [review]
I'd like for this to be added to the docs as well

**Review by Owen1212055 — APPROVED** [review]
Tested, looks good.
Do note that there are some rather simple PRS that it may be nice to squash into this config update.
See: 
https://github.com/PaperMC/Paper/pull/7231
https://github.com/PaperMC/Paper/pull/7622

**Comment by Machine-Maker** [comment]
I think we'll hold off. It's really not an issue to increase the config version frequently.

**Pull request merged**